tcontainer/arrays.go

```diff
@@ -62,9 +60,7 @@ func (s Uint64Slice) Less(i, j int) bool {
 }
 
 func (s Uint64Slice) Swap(i, j int) {
-	s[i] = s[i] ^ s[j]
-	s[j] = s[j] ^ s[i]
-	s[i] = s[i] ^ s[j]
+	s[i], s[j] = s[j], s[i]
 }
```

Comment:
I agree your way is more readable (so I generally prefer it), but the xor version _should_ be faster.
I never measured it though ... will do, then merge.